cli/src/cli_util.rs

```diff
@@ -1837,13 +1861,23 @@ See https://jj-vcs.github.io/jj/latest/working-copy/#stale-working-copy \
             };
         self.user_repo = ReadonlyUserRepo::new(repo);
         let (new_tree_id, stats) = {
-            let mut options = options;
-            let progress = crate::progress::snapshot_progress(ui);
-            options.progress = progress.as_ref().map(|x| x as _);
-            locked_ws
-                .locked_wc()
-                .snapshot(&options)
-                .map_err(snapshot_command_error)?
+            #[cfg(feature = "git")]
+            {
+                let mut options = options;
+                let progress = crate::progress::snapshot_progress(ui);
+                options.progress = progress.as_ref().map(|x| x as _);
+                locked_ws
+                    .locked_wc()
+                    .snapshot(&options)
+                    .map_err(snapshot_command_error)?
+            }
+            #[cfg(not(feature = "git"))]
+            {
+                locked_ws
+                    .locked_wc()
+                    .snapshot(&options)
+                    .map_err(snapshot_command_error)?
+            }
```

Comment:
I couldn't quickly think of a way to write this that didn't have warnings or weird code, but suggestions welcome.

Reply:
I improved this a bit, but I'm not sure I like it.